services/authoring/src/routes/hooks.ts

```diff
@@ -49,7 +49,6 @@ export class HooksController {
     hook: Hook,
   ): Promise<Hook> {
     const hooksRepository = this.hooksRepositoryFactory.createRepository();
-    hook = { keyPath: hook.keyPath, type: hook.type, url: hook.url, tags: hook.tags, format: hook.format };
```

Comment:
This line was meant to prevent extra/garbage fields from being sent in the request and saved (sort of simple validation that just ignores extra fields).
Won't removing this line allow this to happen?

It was also the same in the `PUT` endpoint, which was changed to `...hook`.
I'm not sure how Typescript handles `...hook` when it's of type `Hook`. Does it compile to explicitly spread only the fields from the `Hook` type, or any fields on the object in runtime?

Reply:
It let's any other field come in, which is what @Yshayy and I agreed is best for extensibility